### DISCUSSION TITLE
Incorporating Wind Data to Wind Turbine Models

### DISCUSSION THREAD

**Discussion by TimThacker** [CURENT/andes#377] [Modeling] [answered]
Hantao,

I have been looking into the available wind turbine models in ANDES. Ultimately, my goal is to observe the effects of wind turbine participation in grid frequency support, considering somewhat inconsistent power available corresponding to varying wind speed. 

Reading through the documentation, I have noticed that it is noted multiple times that there are type 3 and type 4 WT models available. The available type 3 model, the WTARA1, does not consider wind speed details. The model that I am assuming is the type 4 model, the WTARV1, is noted to consider wind velocity details. I see a value for wind density, but not wind speed. The documentation notes that the model is a work in progress - could you clarify if this model has been completed or if it is missing the wind velocity I am looking for?

On the note of working with wind turbine models, I believe the WECC system has several wind farms, but I did not see any aerodynamic models in the excel format WECC case. Could you clarify how renewables are considered in the WECC case provided?

Thank you!

**Comment by cuihantao** [comment, marked as answer]
Hi Tim,

`WTARV1` is a custom model that needs to be developed and hasn't been completed. The standard, generic wind models do not include a module for wind speed (as they are meant for transient stability studies). `WTARV1` is supposed to be a drop-in replacement for `WTARA1`. 

If you are interested in contributing, a model to start with in the book "Power System Modelling and Scripting" on page 444. In the same book, wind speed models are developed to provide a velocity input to the aerodynamics model.

**Reply by TimThacker** [reply]
Hantao,

Thank you. I believe my research will require me to use wind turbines (and wind speed data), so I will begin working on a wind turbine model.

**Reply by cuihantao** [reply]
You are welcome. The wind turbine model is there, see: https://www.powerworld.com/files/clientconf2014/05WeberNewWindTurbines.pdf. You will need to develop a model for the wind speed-power characteristics as mentioned above.

**Reply by TimThacker** [reply]
Hantao, I have been looking back into this. I see that aerodynamics gain Ka is listed as a parameter of the WTARA1 model, but I do not see it incorporated into any of the equations. Could you explain how and where it is factored in?

**Reply by barajale** [reply]
Hi, I'm very interested to see if the wind turbine with wind data model has been developed. My research project is interested in using a model just like this. Let me know if I can help with any existing work so I do not need to start from scratch.

**Comment by cuihantao** [comment]
Hi Tim,

It looks like an error on my side. Ka should be multiplied in the output Pmg equations.

Would you like to make a Pull Request for it?

Thanks!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by TimThacker** [reply]
Hantao, I am sorry for the delay. I was busy with the end of the semester and then away for the holidays. I see you have corrected the bug. Thank you!

**Reply by cuihantao** [reply]
You are welcome!
